File: app/src/main/java/codeblock/app/e_news/ApiInterface.java
package codeblock.app.e_news;

import codeblock.app.e_news.Models.Headlines;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiInterface {

    @GET("top-headlines")
    Call<Headlines> getHeadlines(
            @Query("country") String country,
            @Query("apiKey") String apiKey
    );
}

File: app/src/main/java/codeblock/app/e_news/ApiInterface.kt
import codeblock.app.e_news.models.Headlines
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiInterface {
    @GET("everything")
    fun getNewsByQuery(
        @Query("q") query: String?,
        @Query("apiKey") apiKey: String?
    ): Call<Headlines?>?
}
